src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetriesBehavior.cs

```diff
@@ -42,7 +42,8 @@ public override void Invoke(Context context, Action next)
                 {
                     var receiveAddress = PipelineInfo.PublicAddress;
 
-                    var messageToRetry = new OutgoingMessage(context.GetPhysicalMessage().Id, message.Headers, message.Body);
+                    message.RevertToOriginalBodyIfNeeded();
+                    var messageToRetry = new OutgoingMessage(message.Id, message.Headers, message.Body);
```

Comment:
Before we were using `context.GetPhysicalMessage().Id`, now `message.Id`. I like using the `message.Id` better. I assume they should always be the same.